Flashlight-VNC/src/com/flashlight/vnc/VNCClient.as

```diff
@@ -564,13 +562,25 @@ package com.flashlight.vnc
 				logger.info("event.ctrlKey "+event.ctrlKey);
 				logger.info("event.keyLocation "+event.keyLocation);
 				logger.info("event.shiftKey "+event.shiftKey);
-				logger.info("expectKeyInput "+expectKeyInput);
 
 				// Ignore Ctrl-v, onTextInput() handles this
 				if (event.ctrlKey && event.keyCode == Keyboard.V) {
 					return;
 				}
 
+				// Send other ctrl commands
+				if(event.ctrlKey && event.type == flash.events.KeyboardEvent.KEY_DOWN && event.keyCode != 17 && event.keyCode != 16) {
+					var charCode:Number = event.keyCode;
+					if (charCode >= 65 && charCode <= 90 && event.shiftKey != true) {
+						//If it's not shifted we need to unshift the keyCode otherwise VNC server will go insane
```

Comment:
Yeah, qemu has this feature that wants to sync state of capslock and it detects when there's discrepancy between shift-pressedness and capsness of charCode. :)